### PR TITLE
create splash migration store and service file

### PR DESCRIPTION
**Pull request by ykhedher**

### CODE REVIEW

**Comment by ykhedher** [comment]
[Splash Implementation](https://trello.com/c/SZyBRWwS/590-splash-implementation)

File: src/lib/routes/admin-api/user-splash-controller.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    ) {
+        super(config);
+        this.logger = config.getLogger('splash-controller.ts');
+        this.userSplashService = userSplashService;
+
+        this.post('/', this.recordSplash);
+        this.put('/:id', this.updateSplashSettings);
+    }
+
+    private async recordSplash(
```
Comment (FredrikOseberg) [inline src/lib/routes/admin-api/user-splash-controller.ts:32]: This does the same as the /:id endpoint. I think the id endpoint makes more sense in this context.

File: src/lib/routes/admin-api/user-splash-controller.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+            ...req.body,
+            userId: user.id,
+            seen: req.body.seen || false,
+        };
+
+        const updated = await this.userSplashService.updateSplash(splash);
+        res.json(updated);
+    }
+
+    private async updateSplashSettings(
```
Comment (FredrikOseberg) [inline src/lib/routes/admin-api/user-splash-controller.ts:58]: We should return 400 bad request here if seen is not present in the body.

File: src/lib/services/user-splash-service.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        this.userSplashStore = userSplashStore;
+        this.logger = getLogger('services/user-splash-service.js');
+    }
+
+    async getAllUserSplashs(user: User): Promise<Object> {
+        if (user.isAPI) {
+            return [];
+        }
+        try {
+            const splashs = await (
```
Comment (FredrikOseberg) [inline src/lib/services/user-splash-service.ts:28]: Double await?

**Review by FredrikOseberg — CHANGES_REQUESTED** [review]
This looks good! Just a few things from my side before we merge. For example, I don't think it makes sense to have an endpoint to create a splash for a user, because essentially we want to control when we show splash screens - which will require us to run a migration or something similar to update the splash table for all users. 

So I think we just need to option to update a given splash id for a user.

File: src/migrations/20211109103930-add-splash-entry-for-users.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+exports.up = function (db, cb) {
```
Comment (FredrikOseberg) [inline src/migrations/20211109103930-add-splash-entry-for-users.js:1]: @chriswk Can we write this in raw SQL?

Comment (chriswk) [reply src/migrations/20211109103930-add-splash-entry-for-users.js:1]:
```
INSERT INTO user_splash(splash_id, user_id, seen) SELECT 'envrironment', u.id, false FROM users u
```
Should do the trick

**Review by chriswk — CHANGES_REQUESTED** [review]
Added a suggestion for the migration

File: src/migrations/20211109103930-add-splash-entry-for-users.js (hunk 4 above)
Comment (ykhedher) [reply src/migrations/20211109103930-add-splash-entry-for-users.js:1]: Yep thanks 👍🏻

File: src/lib/routes/admin-api/user-splash-controller.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        req: IAuthRequest<any, any, ISplashBody, any>,
+        res: Response,
+    ): Promise<void> {
+        const { user } = req;
+        const { id } = req.params;
+
+        const splash = {
+            ...req.body,
```
Comment (FredrikOseberg) [inline src/lib/routes/admin-api/user-splash-controller.ts:39]: We don't need to accept anything from ```req.body``` here.

**Review by FredrikOseberg — CHANGES_REQUESTED** [review]
Just a small comment here, then I think this is good to go.

File: src/lib/services/user-splash-service.ts [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+                    ...splashObject,
+                    [splash.splashId]: splash.seen,
+                }),
+                {},
+            );
+            return splashs;
+        } catch (err) {
+            this.logger.error(err);
+
+            return [];
```
Comment (FredrikOseberg) [inline src/lib/services/user-splash-service.ts:41]: This function is expected to return an object.

**Review by FredrikOseberg — CHANGES_REQUESTED** [review]

**Review by chriswk — APPROVED** [review]
The migration looks good now, as well as the exists using the select 1 trick is :+1: .

Gogo

**Review by FredrikOseberg — APPROVED** [review]
LGTM

**Pull request merged**